alghaibb/weather-mania
# Added city picker section

# Weather Mania App - Pull Request

## Overview

In this pull request, I introduce several foundational changes to the Weather Mania app, setting the stage for future advanced features and functionalities. The main modifications involve UI enhancements and location selection, adding depth and potential to the app.

## Changes

### 1. Location Selection

I've implemented an initial version of the location selection capability, allowing users to choose a country and city. This sets the groundwork for future accurate weather data delivery based on user-selected locations.

- **Citypicker Component**: A new component, `Citypicker`, has been added, facilitating the user's choice of a country and city. 

### 2. User Interface Improvements

For a more engaging user experience, several UI enhancements have been incorporated:

- A vibrant background gradient, transitioning from `#4b42ee` to `#46abbf`, enhances the main landing page.
- The "Weather Mania" title stands out prominently, immediately capturing attention.
- A subtitle indicating "Powered by OpenAI, Tremor & More!" offers transparency about data sources.
- Elegant dividers and card components provide a polished feel to the overall UI.

### 3. External API Integration

To offer rich location data, I've integrated a library to fetch country and city information:

- **Country and City Data**: By using the "country-state-city" library, the app now has access to updated country and city information.
- **Geographical Data**: With each location selection, pertinent geographical data like latitude, longitude, and more are prepared for subsequent features.

### 4. Navigation Enhancements

Aiming to streamline user experience:

- **Router Integration**: The integration of the Next.js router ensures smooth navigation within the app.
- **Dynamic URLs**: On city selection, the app will prepare a URL containing the latitude and longitude data for future use.

## Future Developments

- **Weather Data Integration**: I plan to integrate real-time weather data APIs to fetch relevant weather information for chosen locations.
- **Backend Development**: The backend functionality for `Citypicker` is on the agenda, ensuring users get the most accurate and timely weather information.
- **User Customization**: Introduction of user preferences and settings for a tailored experience is also in the pipeline.

## How to Test

If you'd like to review the current changes:

1. Clone this pull request.
2. Ensure all required dependencies are in place.
3. Launch the application in the development environment of your choice.

## Feedback and Contributions

I welcome and value feedback from all contributors and users. Whether you're spotting issues, suggesting improvements, or proposing new features, your input will guide the future direction of the Weather Mania app.

Warm regards,
@alghaibb
